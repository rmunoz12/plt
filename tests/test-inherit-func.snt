@setup {

group A() {
    int x;

    func group A __init__() {
        this.x = 1;
        return this;
    }

    func int testA(int x) {
        return x + 5;
    }

    func int testB() {
        return this.x + 20;
    }
};


group B(A) {
    int y;
    int x;

    func group B __init__(int a, int b) {
        this.x = a;
        this.y = b;
        return this;
    }
};


group A obj;
group B obj2;

}

@turns {

func void begin() {
    int y;

    obj.__init__();
    obj2.__init__(2, 3);

    print(obj.x); print("\n");
    print(obj.testA(5)); print("\n");
    print(obj.testB()); print("\n");

    print(obj2.x); print("\n");
    print(obj2.y); print("\n");
    print(obj2.testA(10)); print("\n");
    print(obj2.testB()); print("\n");

    end;
}

}
