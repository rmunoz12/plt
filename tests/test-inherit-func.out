1
10
21
2
3
15
22
